ethcore/src/engines/authority_round/mod.rs

```diff
@@ -59,12 +60,13 @@ mod finality;
 
 /// `AuthorityRound` params.
 pub struct AuthorityRoundParams {
-	/// Time to wait before next block or authority switching,
-	/// in seconds.
+	/// A map defining intervals of blocks with the given times (in seconds) to wait before next
+	/// block or authority switching. The keys in the map are timestamps of starting blocks of those
+	/// periods. The entry at `0` should be defined.
 	///
-	/// Deliberately typed as u16 as too high of a value leads
-	/// to slow block issuance.
-	pub step_duration: u16,
+	/// Wait times (durations) are deliberately typed as `u16` since larger values lead to slow
```

Comment:
We can still store `u64` to avoid casting in the functions that use it and just validate when creating the params or the engine.

Reply:
Yes, we are already checking that the durations are in the range when receiving `AuthorityRoundParams`. I don't mind storing `u64` in the map.

Reply:
But then I wonder whether `u16::MAX` is a good upper bound. This has nothing to do with the number of bits and should just be a threshold above which step durations don't make sense anymore. `u16::MAX` corresponds to about 18 hours. Should we set the limit to 24 hours (86400 seconds)?